doc/manual/pragmas.txt

```diff
@@ -1087,3 +1087,66 @@ In the above example, providing the -d flag causes the symbol
 ``FooBar`` to be overwritten at compile time, printing out 42. If the
 ``-d:FooBar=42`` were to be omitted, the default value of 5 would be
 used.
+
+
+User custom pragmas 
+-------------------
+It is possible to define custom typed pragmas. Custom pragmas do not effect 
+code generation directly, but their presence can be detected by macros.
+Custom pragmas are defined using pragma ``pragma``:
+
+.. code-block:: nim
+  template dbTable(name: string, table_space: string = nil) {.pragma.}
+  template dbKey(name: string = nil, primary_key: bool = false) {.pragma.}
+  template dbForeignKey(t: typedesc) {.pragma.}
+  template dbIgnore {.pragma.}
+
+Consider stylized example of possible Object Relation Mapping (ORM) implementation:
+.. code-block:: nim
+  const tblspace {.strdefine.} = "dev" # switch for dev, test and prod environments 
+
+  type
+    User {.dbTable: ("users", tblspace).} = object
+      id {.dbKey: (primary_key = true).}: int
```

Comment:
Ugh, no. Tuple constructors use ':'!

Reply:
I kind of disagree. Yes, tuple constructors are using ':', but it is not a tuple. It is more of call expression and I am passing the argument by name.
IMO, tuple constructors don't fit as they can't have a mix of unnamed and named arguments, while proc/template calls can.

Let me know if got everything wrong.

Reply:
It's clearly a tuple that is passed to the template here. That said, I think pragmas should just allow ordinary call syntax without the colons, ``importc"foo"`` that would solve this issue.

Reply:
All done now, it well worth a double shot of wiskey at FOSDEM